rust/datafusion/tests/sql.rs

```diff
@@ -910,6 +910,76 @@ async fn csv_query_count_one() {
     assert_eq!(expected, actual);
 }
 
+#[tokio::test]
+async fn case_when() {
+    let mut ctx = ExecutionContext::new();
+    register_aggregate_csv_by_sql(&mut ctx).await;
+    let sql = "SELECT \
+        CASE WHEN c1 = 'a' THEN 1 \
+             WHEN c1 = 'b' THEN 2 \
+             ELSE 999 END \
+        FROM aggregate_test_100 LIMIT 20";
+    let actual = execute(&mut ctx, sql).await;
+    let expected = vec![
+        vec!["999"],
+        vec!["999"],
+        vec!["2"],
+        vec!["1"],
+        vec!["2"],
+        vec!["2"],
+        vec!["999"],
+        vec!["1"],
+        vec!["999"],
+        vec!["1"],
+        vec!["999"],
+        vec!["1"],
+        vec!["999"],
+        vec!["999"],
+        vec!["2"],
+        vec!["999"],
+        vec!["999"],
+        vec!["999"],
+        vec!["999"],
+        vec!["999"],
+    ];
```

Comment:
It is a bit difficult for me to confirm that this is correct. What do you think if we just generate a table in memory with all possibilities and test against that? IMO it is a bit more explicit and also allows to verify the cases with null values.